packages/financial-templates-lib/src/logger/DiscordTransport.ts

```diff
@@ -38,48 +62,63 @@ export class DiscordTransport extends Transport {
     this.postOnNonEscalationPaths = ops.postOnNonEscalationPaths ?? true;
 
     this.logQueue = [];
+    this.isQueueBeingExecuted = false;
+    this.enqueuedLogCounter = 0;
+  }
+
+  // Getter for checking if the transport is flushed.
+  get isFlushed(): boolean {
+    return this.enqueuedLogCounter === 0;
   }
 
   // Note: info must be any because that's what the base class uses.
   async log(info: any, callback: () => void): Promise<void> {
-    try {
```

Comment:
I had to refactor this similar to DiscordTicketTransport in using info type guard and canSend to determine if need to send out the log. This allows avoiding conditional returns that would make tracking enqueued log counter more complex. Now only if we decide log needs to be processed, we increment the counter at the start of try block and decrement when done or caught an error. Apart from additional type checks the logic for deciding when/where to send should be the same as before.